src/cli/repo-checkout.js

```diff
@@ -125,34 +125,47 @@ export default async function main(config, argv) {
     projects: {}
   };
 
-  await Promise.all(projects.map(async (project) => {
-    let projectStart = Date.now();
+  let archivesToExtract = await Promise.all(projects.map(async (project) => {
+    let downloadStart = Date.now();
+    stats.projects[project.name] = {duration: 0};
+
     let repoPath =
       fsPath.join(args.directory, '.repo', 'projects', `${project.path}.git`);
+    let name = `${urlAlias(project.remote)}/${args.branch}`;
+    let namespace = `${args.namespace}.${createHash(name)}`;
+    let archiveDetails = {
+      projectName: project.name
+    };
 
-    // Only attempt to use caches if the project does not exist... Tar will
-    // happy clobber things that likely should not be clobbered if expansion is
-    // triggered from the cache...
+    // Only attempt to use caches if the project does not already exist.
     if (!await fs.exists(repoPath)) {
-      let name = `${urlAlias(project.remote)}/${args.branch}`;
-
-      await artifacts.useIfAvailable(
-        name,
-        `${args.namespace}.${createHash(name)}`,
-        args.directory
-      );
+      archiveDetails.archivePath =
+        await artifacts.downloadIfAvailable(name, namespace, args.directory);
```

Comment:
ah, I was basing this on the previous name and took it as "download it if the cache tarball is available" but I do see the naming being clearer as Unavailable